### PR TITLE
Move lowest inserted body number out of blocktree

### PR DESCRIPTION
**Pull request by asdacap**
- Mode LowestInsertedBodyNumber out of IBlockTree.
- Minor change, just make sense. Era and portal will just make it not very clear
- Delete some interface method from `IBlockTree` and `IBlockStore`. 
- LowestInsertedReceipt is not in IBlockTree at all. 

#### What types of changes does your code introduce?

- [x] Refactoring

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [X] Yes
- [ ] No

#### Notes on testing

- Testing...

### CODE REVIEW

**Review by benaadams — APPROVED** [review]

File: src/Nethermind/Nethermind.Api/NethermindApi.cs [hunk 1]
```diff
@@ -226,6 +226,7 @@ public ISealEngine SealEngine
         public CensorshipDetector CensorshipDetector { get; set; } = null!;
         public IWallet? Wallet { get; set; }
         public IBadBlockStore? BadBlocksStore { get; set; }
+        public BlockStore? BlockStore { get; set; }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Api/NethermindApi.cs:229]: Shouldn't it be `IBlockStore` in the API?

File: src/Nethermind/Nethermind.Blockchain/Blocks/IBlockStore.cs [hunk 2]
```diff
@@ -19,9 +19,6 @@ public interface IBlockStore
     byte[]? GetRlp(long blockNumber, Hash256 blockHash);
     ReceiptRecoveryBlock? GetReceiptRecoveryBlock(long blockNumber, Hash256 blockHash);
     void Cache(Block block);
-
-    // These two are used by blocktree. Try not to use them...
-    void SetMetadata(byte[] key, byte[] value);
-    byte[]? GetMetadata(byte[] key);
     bool HasBlock(long blockNumber, Hash256 blockHash);
+    long? LowestInsertedBodyNumber { get; set; }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Blockchain/Blocks/IBlockStore.cs:23]: Does it need public set? Can it just update it internally on `Insert`?

Comment (asdacap) [reply src/Nethermind/Nethermind.Blockchain/Blocks/IBlockStore.cs]: Can't as erafiles and portal would also use insert.

File: src/Nethermind/Nethermind.Api/NethermindApi.cs (hunk 1 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Api/NethermindApi.cs]: Moved to ISyncPointers.

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**